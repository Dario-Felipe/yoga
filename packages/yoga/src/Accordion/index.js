import Accordion from './web';

export default Accordion;
