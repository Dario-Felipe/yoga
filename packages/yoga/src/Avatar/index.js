import Avatar from './web/Avatar';
import AvatarCircle from './web/AvatarCircle';

Avatar.Circle = AvatarCircle;

export default Avatar;
