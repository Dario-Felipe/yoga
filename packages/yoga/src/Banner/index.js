import Banner from './web';

export default Banner;
