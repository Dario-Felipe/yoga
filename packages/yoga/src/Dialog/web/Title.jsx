import styled from 'styled-components';

import { Card } from '../../Card';

export const Header = styled(Card.Header)`
  padding: 0;
  margin: 0;
`;
