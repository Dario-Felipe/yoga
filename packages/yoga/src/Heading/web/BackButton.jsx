import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { ArrowLeft } from '@gympass/yoga-icons';
import Button from '../../Button';
import Box from '../../Box';

const ButtonIcon = styled(Button.Icon)`
  ${({
    theme: {
      yoga: {
        components: { heading },
      },
    },
  }) => {
    return `
    width: ${heading.button.width}px;
    height: ${heading.button.height}px;
  `;
  }}
`;

const BackButton = ({ onClick, ...props }) => (
  <Box padding="xxxsmall" onClick={onClick} role="button" tabIndex={0}>
    <ButtonIcon tabIndex={-1} icon={ArrowLeft} inverted secondary {...props} />
  </Box>
);

BackButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

BackButton.displayName = 'Heading.BackButton';

export default BackButton;
