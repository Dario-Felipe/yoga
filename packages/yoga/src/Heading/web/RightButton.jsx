import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import Button from '../../Button';

const ButtonIcon = styled(Button.Icon)`
  ${({
    secondary,
    theme: {
      yoga: {
        components: { heading },
      },
    },
  }) =>
    css`
      width: ${heading.button.width}px;
      height: ${heading.button.height}px;
      background-color: ${secondary
        ? 'transparent'
        : heading.button.background};

      :not(:last-of-type) {
        margin-right: ${heading.button.marginRight}px;
      }
    `};
`;

const RightButton = ({ onClick, icon, ...props }) => (
  <ButtonIcon icon={icon} secondary inverted onClick={onClick} {...props} />
);

RightButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  icon: PropTypes.elementType.isRequired,
};

RightButton.displayName = 'Heading.RightButton';

export default RightButton;
