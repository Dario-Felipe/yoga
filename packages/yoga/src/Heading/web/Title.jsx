import React from 'react';
import PropTypes from 'prop-types';
import Text from '../../Text';
import Box from '../../Box';

const Title = ({ children, ...props }) => (
  <Box w="100%" textAlign="center">
    <Text.Body1 {...props} fontSize="small" as="h1" bold>
      {children}
    </Text.Body1>
  </Box>
);

Title.propTypes = {
  children: PropTypes.string.isRequired,
};

Title.displayName = 'Heading.Title';

export default Title;
