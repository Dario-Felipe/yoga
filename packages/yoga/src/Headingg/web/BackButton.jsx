import React from 'react';
import PropTypes from 'prop-types';
import { ArrowLeft } from '@gympass/yoga-icons';
import Button from '../../Button';

const BackButton = ({ onClick }) => (
  <Button.Icon icon={ArrowLeft} inverted onClick={onClick} />
);

BackButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default BackButton;
