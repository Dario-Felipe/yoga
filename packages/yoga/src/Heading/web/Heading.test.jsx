import React from 'react';
import { render } from '@testing-library/react';

import { Upload2 } from '@gympass/yoga-icons/src';
import { ThemeProvider, Heading } from '../..';
import Title from './Title';
import BackButton from './BackButton';
import RightButton from './RightButton';

const onClick = () => {};

describe('<Heading />', () => {
  it('should match snapshot with title', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading>
          <Title>Gympass</Title>
        </Heading>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with title and back button', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading>
          <Title>Gympass</Title>
          <BackButton onClick={onClick} />
        </Heading>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with all components', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading>
          <Title>Gympass</Title>
          <BackButton onClick={onClick} />
          <RightButton icon={Upload2} onClick={onClick} />
        </Heading>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with back button', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading>
          <BackButton onClick={onClick} />
        </Heading>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot no padding', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading noPadding>
          <Title>Gympass</Title>
          <BackButton onClick={onClick} />
          <RightButton icon={Upload2} onClick={onClick} />
        </Heading>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
