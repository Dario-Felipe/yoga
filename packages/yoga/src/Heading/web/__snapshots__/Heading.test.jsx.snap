// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot with all components 1`] = `
.c3 {
  width: 100%;
  text-align: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 24px;
  height: 24px;
}

.c1 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #D8385E;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #FFFFFF;
  color: #D8385E;
  padding: 0;
  width: 48px;
}

.c1 svg {
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(216,56,94,0.45);
}

.c1:active {
  background-color: rgba(216,56,94,0.75);
  color: #FFFFFF;
}

.c1:active svg {
  fill: #FFFFFF;
}

.c1 svg {
  fill: #D8385E;
}

.c1:active {
  background-color: rgba(255,255,255,0.75);
  color: rgba(216,56,94,0.75);
}

.c1:active svg {
  fill: rgba(216,56,94,0.75);
}

.c1:not([disabled]):hover,
.c1:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(255,255,255,0.45);
}

.c1 svg {
  width: unset;
  height: unset;
  margin-right: unset;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  font-weight: 500;
  font-family: Rubik;
  color: #231B22;
}

.c0 {
  background: #FFFFFF;
  padding: 16px;
  width: 100%;
  min-height: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:769px) {
  .c0 {
    padding: 16px 72px;
  }
}

<div>
  <header
    class="c0"
  >
    <button
      class="c1"
    >
      <svg
        class="c2"
        height="24"
        width="24"
      />
    </button>
    <div
      class="c3"
    >
      <h1
        class="c4"
      >
        Gympass
      </h1>
    </div>
    <div
      class="c5"
      display="flex"
    >
      <button
        class="c1"
      >
        <svg
          class="c2"
          height="24"
          width="24"
        />
      </button>
    </div>
  </header>
</div>
`;

exports[`<Heading /> should match snapshot with back button 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 24px;
  height: 24px;
}

.c3 {
  width: 100%;
}

.c1 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #D8385E;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #FFFFFF;
  color: #D8385E;
  padding: 0;
  width: 48px;
}

.c1 svg {
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(216,56,94,0.45);
}

.c1:active {
  background-color: rgba(216,56,94,0.75);
  color: #FFFFFF;
}

.c1:active svg {
  fill: #FFFFFF;
}

.c1 svg {
  fill: #D8385E;
}

.c1:active {
  background-color: rgba(255,255,255,0.75);
  color: rgba(216,56,94,0.75);
}

.c1:active svg {
  fill: rgba(216,56,94,0.75);
}

.c1:not([disabled]):hover,
.c1:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(255,255,255,0.45);
}

.c1 svg {
  width: unset;
  height: unset;
  margin-right: unset;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c0 {
  background: #FFFFFF;
  padding: 16px;
  width: 100%;
  min-height: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  width: 48px;
  height: 40px;
}

@media (min-width:769px) {
  .c0 {
    padding: 16px 72px;
  }
}

<div>
  <header
    class="c0"
  >
    <button
      class="c1"
    >
      <svg
        class="c2"
        height="24"
        width="24"
      />
    </button>
    <div
      class="c3"
    />
    <div
      class="c4"
      display="flex"
    >
      <div
        class="c5"
      />
    </div>
  </header>
</div>
`;

exports[`<Heading /> should match snapshot with title 1`] = `
.c2 {
  width: 100%;
  text-align: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  font-weight: 500;
  font-family: Rubik;
  color: #231B22;
}

.c0 {
  background: #FFFFFF;
  padding: 16px;
  width: 100%;
  min-height: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 48px;
  height: 40px;
}

@media (min-width:769px) {
  .c0 {
    padding: 16px 72px;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    >
      <h1
        class="c3"
      >
        Gympass
      </h1>
    </div>
    <div
      class="c4"
      display="flex"
    >
      <div
        class="c1"
      />
    </div>
  </header>
</div>
`;

exports[`<Heading /> should match snapshot with title and back button 1`] = `
.c3 {
  width: 100%;
  text-align: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 24px;
  height: 24px;
}

.c1 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #D8385E;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #FFFFFF;
  color: #D8385E;
  padding: 0;
  width: 48px;
}

.c1 svg {
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(216,56,94,0.45);
}

.c1:active {
  background-color: rgba(216,56,94,0.75);
  color: #FFFFFF;
}

.c1:active svg {
  fill: #FFFFFF;
}

.c1 svg {
  fill: #D8385E;
}

.c1:active {
  background-color: rgba(255,255,255,0.75);
  color: rgba(216,56,94,0.75);
}

.c1:active svg {
  fill: rgba(216,56,94,0.75);
}

.c1:not([disabled]):hover,
.c1:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(255,255,255,0.45);
}

.c1 svg {
  width: unset;
  height: unset;
  margin-right: unset;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  font-weight: 500;
  font-family: Rubik;
  color: #231B22;
}

.c0 {
  background: #FFFFFF;
  padding: 16px;
  width: 100%;
  min-height: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  width: 48px;
  height: 40px;
}

@media (min-width:769px) {
  .c0 {
    padding: 16px 72px;
  }
}

<div>
  <header
    class="c0"
  >
    <button
      class="c1"
    >
      <svg
        class="c2"
        height="24"
        width="24"
      />
    </button>
    <div
      class="c3"
    >
      <h1
        class="c4"
      >
        Gympass
      </h1>
    </div>
    <div
      class="c5"
      display="flex"
    >
      <div
        class="c6"
      />
    </div>
  </header>
</div>
`;
